File: leetcode/0463. Island Perimeter/Solution.swift
// https://leetcode.com/problems/island-perimeter/

class Solution {
    func islandPerimeter(_ grid: [[Int]]) -> Int {
        var land = 0, adjacent = 0
        for i in 0..<grid.count {
            for j in 0..<grid[0].count {
                if grid[i][j] == 1 {
                    land += 1
                    if i + 1 < grid.count && grid[i + 1][j] == 1 {
                        adjacent += 1
                    }
                    if j + 1 < grid[0].count && grid[i][j + 1] == 1 {
                        adjacent += 1
                    }
                }
            }
        }
        
        return 4 * land - 2 * adjacent
    }
}

File: leetcode/0463. Island Perimeter/Solution.py
# https://leetcode.com/problems/island-perimeter/

from typing import List

class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        land = 0
        adjacent = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    land += 1
                    if i + 1 < len(grid) and grid[i + 1][j] == 1:
                        adjacent += 1
                    if j + 1 < len(grid[0]) and grid[i][j + 1] == 1:
                        adjacent += 1
        
        return 4 * land - 2 * adjacent

s = Solution()
assert s.islandPerimeter([[0,1,0,0],[1,1,1,0],[0,1,0,0],[1,1,0,0]]) == 16